simde/x86/avx512/dpbusd.h

```diff
@@ -0,0 +1,55 @@
+#if !defined(SIMDE_X86_AVX512_DPBUSD_H)
+#define SIMDE_X86_AVX512_DPBUSD_H
+
+#include "types.h"
+
+HEDLEY_DIAGNOSTIC_PUSH
+SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
+SIMDE_BEGIN_DECLS_
+
+SIMDE_FUNCTION_ATTRIBUTES
+simde__m128i
+simde_mm_dpbusd_epi32(simde__m128i src, simde__m128i a, simde__m128i b) {
+  #if defined(SIMDE_X86_AVX512VL_NATIVE) && defined(SIMDE_X86_AVX512VNNI_NATIVE)
+    return _mm_dpbusd_epi32(src, a, b);
+  #else
+    simde__m128i_private
+      src_ = simde__m128i_to_private(src),
+      a_ = simde__m128i_to_private(a),
+      b_ = simde__m128i_to_private(b);
+    #if defined(SIMDE_SHUFFLE_VECTOR_) && defined(SIMDE_CONVERT_VECTOR_)
+      simde__m512i_private
+        r1_,
+        r2_;
+
+      a_.u8 = SIMDE_SHUFFLE_VECTOR_(8, 16, a_.u8, a_.u8, 0, 4, 8, 12, 1, 5, 9, 13, 2, 6, 10, 14, 3, 7, 11, 15);
+      b_.i8 = SIMDE_SHUFFLE_VECTOR_(8, 16, b_.i8, b_.i8, 16, 20, 24, 28, 17, 21, 25, 29, 18, 22, 26, 30, 19, 23, 27, 31);
+
+      SIMDE_CONVERT_VECTOR_(r1_.u32, a_.u8);
+      SIMDE_CONVERT_VECTOR_(r2_.i32, b_.i8);
+
+      src_.i32 =
+        src_.i32 +
```

Comment:
I think we discussed (on our call on Saturday) using `src_i32 += ...` instead of `src_.i32 = src_.i32 + ...` and you said you had tried but it didn't work somewhere; did you remember where that was?  There should probably be a comment (at least on the first implementation) explaining the issue since IMHO it's a bit of a code smell.

Reply:
Made the changes. I must have been doing something else wrong when I tried += before. It's working now.